src/Hamiltonians/ExtendedHubbardReal1D.jl

```diff
@@ -86,21 +89,30 @@ function Base.getproperty(
 ) where BOUNDARY_CONDITION
     BOUNDARY_CONDITION
 end
+function Base.getproperty(
+    ::ExtendedHubbardReal1D{<:Any,<:Any,<:Any,<:Any,<:Any,<:Any,INTERACTION},
+    ::Val{:interaction}
+) where INTERACTION
+    INTERACTION
+end
 
 function num_offdiagonals(::ExtendedHubbardReal1D, address::SingleComponentFockAddress)
     return 2 * num_occupied_modes(address)
 end
 
 """
-    extended_hubbard_interaction(h::ExtendedHubbardReal1D, address)
+    extended_hubbard_interaction(h::ExtendedHubbardReal1D, address, interaction)
 
-Compute and return both the nearest neighbor occupation number product
-``\\sum_j n_j n_{j+1}`` (according to the boundary conditions of `h`) as well as the on-site
+Compute and return both the extended range occupation number product
+``\\sum_j f_{i-j} n_i n_{j}`` (according to the boundary conditions of `h`) as well as the on-site
 product ``\\sum_j n_j (n_j - 1)`` treating the `address` as a one-dimensional chain.
 
+where ``f_{i-j} = 1`` for nearest neighbors (interaction = :nearest_neighbour) and 
+``f_{i-j} = (|i-j|)^{-interaction}`` for inverse distance interaction (interaction = <:Number).
+
 See [`ExtendedHubbardReal1D`](@ref) and [`hopnextneighbour`](@ref).
 """
-function extended_hubbard_interaction(h::ExtendedHubbardReal1D, b::SingleComponentFockAddress)
+function extended_hubbard_interaction(h::ExtendedHubbardReal1D, b::SingleComponentFockAddress, _::Symbol)
```

Comment:
```suggestion
function extended_hubbard_interaction(h::ExtendedHubbardReal1D, b::SingleComponentFockAddress, ::Symbol)
```
Don't need the `_` here.

Reply:
Changes have been made.